web_monitoring/html_diff_render.py

```diff
@@ -536,6 +549,16 @@ def __new__(cls, tag, data, html_repr, pre_tags=None,
         obj.html_repr = html_repr
         return obj
 
+    def __eq__(self, other):
+        # This equality check aims to apply specific rules to the contents
+        # of the tag element solving false positive cases
+        if Strict_url_rule.mode:
+            return clean_resource_url(self, other)
```

Comment:
I think this might wind up being way too broad. First, if `other` is a normal word in the page’s text (perhaps a URL that is fully written out instead of an actual element), you’re going to wind up potentially considering this tag equal to it, which clearly isn’t right.

You should probably start by making sure `other` is also an instance of `tag_token`.

Then it might be a good idea to try and pull out all the URLs or URL-containing attributes to compare. For example, an `<img ...>` tag might have `src`, but it might also have `srcset`.

Then, since this is a generic comparison instead of one specific to `<img ...>` tags, you should probably also compare all the other attributes if the URL-containing ones are equal. (Alternatively, you could create a new `ImgToken` type that inherits from `tag_token` and is just for `<img ...>` tags and implement `__eq__` on that instead.)